eslint/eslintrc-shared.yml

```diff
@@ -15,38 +18,136 @@ extends:
 rules:
     # 2 means error, 1 means warning, 0 means ignore.
 
+    # Four-space indent.
+    # Note: Some problems with const.
+    # indent: ["error", 4]
+    no-mixed-spaces-and-tabs: "error"
+    no-trailing-spaces: "error"
+
+    # Newlint at end of file.
+    eol-last: "error"
+
+    # Line comments start with a space <-- like this one does.
+    spaced-comment: "error"
+
+    # var compositeThing = new CompositeThing();
+    # var params = { thisIsPreferred: true };
+    # var params = { this_is_okay: true };
+    camelcase: ["error", { "properties": "never" }]
+    new-cap: "error"
+
+    # if (cond) {
+    #     stuff();
+    # } else {
+    #     otherStuff();
+    # }
+    # for (var i = 0; i < target; ++i) {
+    #     stuff();
+    # };
+    # var foo = function (arg1, arg2) {
+    #     stuff(arg1);
+    # };
+    # function factorial (value) {
+    #     return value == 0 ? 1 : value * factorial(value - 1);
+    # };
+    keyword-spacing: ["error", { "before": true, "after": true }]
+    brace-style: ["error", "1tbs", { "allowSingleLine": true }]
+    space-before-blocks: ["error", "always"]
+    curly: "error"
+    semi-spacing: "error"
+    func-style: "error"
```

Comment:
Hmm, blue-frontend is generating some errors on `func-style`, but they are on non-expression functions. In general the arrow functions seem to be okay, at least the anonymous arrow functions.

We should look through those errors though.

I've been wanting to switch away from expression functions which has nicer aesthetics with ES6. Maybe this is the right time.